macros/src/lib.rs

```diff
@@ -29,6 +33,25 @@ pub fn test(
     }
 }
 
+/// This macro generates a type tag for a struct.
+/// It creates a new enum with a single variant that is used as a type tag.
+/// The type tag is used for serialization and deserialization as well as for schema generation.
```

Comment:
mabye drop the term "OpenApi" here or give a usage example.

Reply:
[a0d6d78](https://github.com/geo-engine/geoengine/pull/1013/commits/a0d6d789b320cef60ec36807ab6f178bd6b38edd)